Stratis.Bitcoin.Features.Miner/PowMining.cs

```diff
@@ -155,24 +155,24 @@ public List<uint256> GenerateBlocks(ReserveScript reserveScript, ulong generate,
                     if (newChain.ChainWork <= chainTip.ChainWork)
                         continue;
 
-                    this.chain.SetTip(newChain);
+                    var blockResult = new BlockItem { Block = pblock };
 
-                    var blockResult = new BlockResult { Block = pblock };
-                    this.consensusLoop.AcceptBlock(new ContextInformation(blockResult, this.network.Consensus));
-                    this.consensusLoop.Puller.SetLocation(newChain);
+                    this.consensusLoop.AcceptBlock(blockResult);
 
                     if (blockResult.ChainedBlock == null)
-                        break; // Reorg.
-
-                    if (blockResult.Error != null)
+                    {
+                        this.logger.LogTrace("(-)[REORG-2]");
                         return blocks;
+                    }
 
-                    // Push the block to disk, so it is available when peers ask for it.
-                    this.blockRepository.PutAsync(blockResult.ChainedBlock.HashBlock, new List<Block> { pblock }).GetAwaiter().GetResult();
+                    if (blockResult.Error != null)
+                    {
+                        if (blockResult.Error == ConsensusErrors.InvalidPrevTip)
```

Comment:
this can happen , but just below, we have catch clause that catches this very consensus error and imho that can't happen, so you can maybe completely remove this try/catch block

Reply:
Good point